agent/agent_test.go

```diff
@@ -15,17 +15,11 @@ func TestNewLogAgent(t *testing.T) {
 	mockCfg := Config{}
 	mockLogger := zap.NewNop().Sugar()
 	mockPluginDir := "/some/path/plugins"
-	mockDatabaseFile := "/some/path/database"
-	mockParameterKey := "test"
-	mockParameterValue := "value"
-	agent := NewLogAgent(&mockCfg, mockLogger, mockPluginDir, mockDatabaseFile).
-		WithBuildParameter(mockParameterKey, mockParameterValue)
+	mockDatabaseFile := ""
+	agent, err := NewLogAgent(&mockCfg, mockLogger, mockPluginDir, mockDatabaseFile, nil)
```

Comment:
Why take the build parameter test out?

Reply:
See below for more on the interface change, but for the test, it's consumed inside the function, so it doesn't test any additional surface area becasue we aren't building any operators in this test anyways.